src/components/NewCommentForm.tsx

```diff
@@ -75,25 +191,40 @@ export const NewCommentForm: React.FC = () => {
             id="comment-body"
             name="body"
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={classNames('textarea', {
+              'is-danger': !hasComment,
+            })}
+            value={comment}
+            onChange={handleCommentChange}
           />
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Enter some text
-        </p>
+        {!hasComment && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            {CommentFormErrors.COMMENT}
+          </p>
+        )}
       </div>
 
       <div className="field is-grouped">
         <div className="control">
-          <button type="submit" className="button is-link is-loading">
+          <button
+            type="submit"
+            className={classNames('button', 'is-link', {
+              'is-loading': isCommentAdding,
+            })}
+            onClick={handleFormSubmit}
+          >
```

Comment:
onClick event on a submit button is unnecessary. The submit event is automatically triggered when the button of type 'submit' is clicked. So, handleFormSubmit can be removed from onClick event.